packages/amplify-provider-awscloudformation/src/network/stack.ts

```diff
@@ -0,0 +1,196 @@
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+
+    const cfn = this._toCloudFormation();
+
+    return cfn;
+  }
+}
+
+function createVpc(scope: cdk.Construct, vpcId: string, vpcName: string, internetGatewayId: string) {
+  const vpcCidrBlock = '10.0.0.0/16';
+  const condition = new cdk.CfnCondition(scope, 'UseNewVpcCondition', {
+    expression: cdk.Fn.conditionAnd(cdk.Fn.conditionEquals(vpcId, ''), cdk.Fn.conditionEquals(internetGatewayId, '')),
+  });
+
+  const vpc = new ec2.Vpc(scope, vpcName, {
+    cidr: vpcCidrBlock,
+    subnetConfiguration: [],
+  });
+  (vpc.node.defaultChild as ec2.CfnVPC).tags.setTag('Name', vpcName);
+
+  const cfnVpc = <ec2.CfnVPC>vpc.node.defaultChild;
+  cfnVpc.cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.RETAIN;
+  cfnVpc.cfnOptions.condition = condition;
+
+  const outputVpc = cdk.Fn.conditionIf(condition.logicalId, vpc.vpcId, vpcId);
+  new cdk.CfnOutput(scope, 'VpcId', {
+    value: outputVpc as any,
+  });
+  new cdk.CfnOutput(scope, 'VpcCidrBlock', {
+    value: vpcCidrBlock as any,
+  });
+
+  const igw = new ec2.CfnInternetGateway(scope, 'InternetGateway');
+  igw.tags.setTag('Name', `${scope.node.id}/${igw.logicalId}`);
+  igw.cfnOptions.condition = condition;
+  igw.cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.RETAIN;
+
+  const outputIgw = cdk.Fn.conditionIf(condition.logicalId, igw.ref, internetGatewayId);
+  new cdk.CfnOutput(scope, 'Igw', {
+    value: outputIgw as any,
+  });
+
+  const att = new ec2.CfnVPCGatewayAttachment(scope, 'VPCGatewayAttachment', {
+    vpcId: vpc.vpcId,
+    internetGatewayId: igw.ref,
+  });
+  att.cfnOptions.condition = condition;
+  att.cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.RETAIN;
+
+  return {
+    outputVpc,
+    outputIgw,
+    vpcCidrBlock,
+  };
+}
+
+function createAmplifyEnv(
+  scope: cdk.Construct,
+  envName: string,
+  vpcId: string,
+  vpcCidrBlock: string,
+  igwId: string,
+  subnetCidrs: ReadonlyMap<string, string>,
+) {
+  const availabilityZones = [];
+
+  const azSubnetMap = new cdk.CfnMapping(scope, 'AzsMap');
+  subnetCidrs.forEach((cidr, az) => {
+    availabilityZones.push(az);
+    azSubnetMap.setValue(az, 'SubnetCidrBlock', cidr);
+  });
+
+  const vpc = ec2.Vpc.fromVpcAttributes(scope, 'vpc', { vpcId, availabilityZones });
+
+  const cluster = new ecs.CfnCluster(scope, 'Cluster');
+
+  new cdk.CfnOutput(scope, 'ClusterName', {
+    value: cdk.Fn.ref(cluster.logicalId),
+  });
+
+  const subnets: ec2.ISubnet[] = [];
+
+  const pubNacl = new ec2.NetworkAcl(scope, 'Nacl', {
+    vpc,
+  });
+  pubNacl.addEntry('Egress', {
+    cidr: ec2.AclCidr.anyIpv4(),
+    ruleNumber: 100,
+    traffic: ec2.AclTraffic.allTraffic(),
+    direction: ec2.TrafficDirection.EGRESS,
+    ruleAction: ec2.Action.ALLOW,
+  });
+
+  pubNacl.addEntry(`IngressVPC`, {
+    cidr: ec2.AclCidr.ipv4(vpcCidrBlock),
+    ruleNumber: 200,
+    traffic: ec2.AclTraffic.allTraffic(),
```

Comment:
I guess we cannot be more strict here based on the power numbers exposed by the configured containers?